app/controllers/users_controller.rb

```diff
@@ -1,3 +1,30 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:error] = t "user.show.not_found"
+    redirect_to root_path
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = t "user.new.signup_success"
+      redirect_to @user
+    else
+      flash[:fail] = t "user.new.signup_fail"
```

Comment:
```suggestion
      flash.now[:fail] = t "user.new.signup_fail"
```
ri nhé, vì chỗ này chỉ render thôi